src/nvim/eval.c

```diff
@@ -7981,6 +7986,110 @@ static void f_argv(typval_T *argvars, typval_T *rettv)
   }
 }
 
+// Prepare "gap" for an assert error and add the sourcing position.
+static void prepare_assert_error(garray_T *gap)
+{
+  char buf[NUMBUFLEN];
+
+  ga_init(gap, 1, 100);
+  if (sourcing_name != NULL) {
+    ga_concat(gap, sourcing_name);
+    if (sourcing_lnum > 0) {
+      ga_concat(gap, (char_u *)" ");
+    }
+  }
+  if (sourcing_lnum > 0) {
+    vim_snprintf(buf, ARRAY_SIZE(buf), "line %ld", (long)sourcing_lnum);
```

Comment:
`%ld` should be a portable format specifier, see https://github.com/neovim/neovim/wiki/Merging-patches-from-upstream-Vim#code-differences .

I will make this change.